### PR TITLE
T-17793: V1 test adjustment

### PR DESCRIPTION
**Pull request by ricantar**
- Update test for new v1 verstion

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -81,6 +81,8 @@ func NewRouter(reader driver.Reader, writer driver.Writer, apiKeys map[string]bo
 	rt.Router.HandleFunc("/v1/pay_plan", rt.GetPayPlans).Methods(http.MethodGet)
 	rt.Router.HandleFunc("/v1/pay_plan/{type}", rt.GetPayPlan).Methods(http.MethodGet)
 
+	rt.Router.HandleFunc("/v1/redirect", rt.CreateRedirect).Methods(http.MethodPost)
```
Comment (commoddity) [inline router/router.go:84]: Just a quick FYI on this one: I changed this endpoint to `/v1/blockchain/redirect` as it should have been that from the start, so this endpoint won't be needed. 

There's just one place that this needs to be updated which is the `new-chains-ci` repo.

File: cache/mock.go [hunk 2]
```diff
@@ -29,3 +30,33 @@ func NewReaderMock(t *testing.T) *ReaderMock {
 func (r *ReaderMock) NotificationChannel() <-chan *types.Notification {
 	return r.notification
 }
+
```
Comment (commoddity) [inline cache/mock.go:33]: These methods shouldn't be needed as they already exists on the `driver.MockDriver` struct provided by the `portal-db` repo.

File: tests/e2e_testdata.go [hunk 3]
```diff
@@ -1,5 +1,3 @@
-//go:build tests
```
Comment (commoddity) [inline tests/e2e_testdata.go:1]: The build tag should stay as we don't want this data to get compiled into the final binary.

**Comment by commoddity** [comment]
Looks good aside from the comment I already made in the changed files. Thank you for this PR.

File: router/router.go (hunk 1 above)
Comment (ricantar) [reply router/router.go:84]: Good. Thank you for letting me know

**Review by commoddity — CHANGES_REQUESTED** [review]
Left some comments for requested changes before we can merge this one.

File: cache/mock.go (hunk 2 above)
Comment (ricantar) [reply cache/mock.go:33]: Unfortunately, we need it as the driver uses a different implementation and doesn't work (tried and doesn't work)

Comment (commoddity) [reply cache/mock.go:33]: I just tested on my local workstation ad seem to work fine without them. I've pushed a few updates I had to make to the test schema get it working on this branch.

**Comment by commoddity** [comment]
Pushed a few updates to the DB schema to get the test containers to start up correctly and added the `tests` build tag to get the new `v1` tests to run. 

Unfortunately it seems like the tests that hit the V1 endpoints are not working (failing on the `update` applications one). Let's dig into that on Monday though; signing off for the weekend.

**Pull request merged**